core/storage/statistics/gae_models.py

```diff
@@ -59,6 +59,183 @@ def get_or_create(cls, exploration_id, state_name):
         return counter
 
 
+class AnswerSubmittedEventLogEntryModel(base_models.BaseModel):
+    """An event triggered by a student submitting an answer.
+
+    Event schema documentation
+    --------------------------
+    V1:
+        event_type: 'answer_submit'
+        exp_id: id of exploration currently being played
+        exp_version: version of exploration
+        state_name: Name of current state
+        client_time_spent_in_secs: Time since start of this state when this
+            event was recorded
+        session_id: ID of current student's session
+        is_feedback_useful: Whether the answer received useful feedback
+        is_answer_correct: Whether the answer is correct
+    """
+    # Which specific type of event this is
+    event_type = ndb.StringProperty(indexed=True)
+    # Id of exploration currently being played.
+    exp_id = ndb.StringProperty(indexed=True)
+    # Current version of exploration.
+    exp_version = ndb.IntegerProperty(indexed=True)
+    # Name of current state.
+    state_name = ndb.StringProperty(indexed=True)
+    # ID of current student's session
+    session_id = ndb.StringProperty(indexed=True)
+    # Time since start of this state before this event occurred (in sec).
+    client_time_spent_in_secs = ndb.FloatProperty(indexed=True)
+    # Whether the submitted answer received useful feedback
+    is_feedback_useful = ndb.BooleanProperty(indexed=True)
+    # Whether the submitted answer is the correct answer.
+    is_answer_correct = ndb.BooleanProperty(indexed=True)
```

Comment:
do you need index on every attribute?

(indexing usually involves computation and re indexing on insert deletes. So its better not to index an attribute if you are sure that you won't need to query data by that.)

Reply:
@1995YogeshSharma We need index on this attribute. In milestone 4, we plan to implement a MR job that can recompute statistics based on event models. There, we need to retrieve event models and query by the boolean fields.